modules/asset-manager-impl/src/main/java/org/opencastproject/assetmanager/impl/AbstractAssetManagerWithTieredStorage.java

```diff
@@ -0,0 +1,412 @@
+      throw new NotFoundException("Mediapackage " + mpId + " not found!");
+    }
+
+    processOperations(results, targetStore);
+  }
+
+  public RichAResult getSnapshotsByIdAndVersion(final String mpId, final Version version) {
+    RequireUtil.requireNotBlank(mpId, "mpId");
+    RequireUtil.notNull(version, "version");
+    AQueryBuilder q = createQuery();
+    ASelectQuery query = baseQuery(q, version, mpId);
+    return Enrichments.enrich(query.run());
+  }
+
+  public void moveSnapshotsByIdAndVersion(final String mpId, final Version version, final String targetStore)
+          throws NotFoundException {
+    RichAResult results = getSnapshotsByIdAndVersion(mpId, version);
+
+    if (results.getRecords().isEmpty()) {
+      throw new NotFoundException("Mediapackage " + mpId + "@" + version.toString() + " not found!");
+    }
+
+    processOperations(results, targetStore);
+  }
+
+  public RichAResult getSnapshotsByDate(final Date start, final Date end) {
+    RequireUtil.notNull(start, "start");
+    RequireUtil.notNull(end, "end");
+    AQueryBuilder q = createQuery();
+    ASelectQuery query = baseQuery(q).where(q.archived().ge(start)).where(q.archived().le(end));
+    return Enrichments.enrich(query.run());
+  }
+
+  public void moveSnapshotsByDate(final Date start, final Date end, final String targetStore)
+          throws NotFoundException {
+    RichAResult results = getSnapshotsByDate(start, end);
+
+    if (results.getRecords().isEmpty()) {
+      throw new NotFoundException("No media packages found between " + start + " and " + end);
+    }
+
+    processOperations(results, targetStore);
+  }
+
+  public RichAResult getSnapshotsByIdAndDate(final String mpId, final Date start, final Date end) {
+    RequireUtil.requireNotBlank(mpId, "mpId");
+    RequireUtil.notNull(start, "start");
+    RequireUtil.notNull(end, "end");
+    AQueryBuilder q = createQuery();
+    ASelectQuery query = baseQuery(q, mpId).where(q.archived().ge(start)).where(q.archived().le(end));
+    return Enrichments.enrich(query.run());
+  }
+
+  public void moveSnapshotsByIdAndDate(final String mpId, final Date start, final Date end, final String targetStore)
+          throws NotFoundException {
+    RichAResult results = getSnapshotsByDate(start, end);
+
+    if (results.getRecords().isEmpty()) {
+      throw new NotFoundException("No media package with id " + mpId + " found between " + start + " and " + end);
+    }
+
+    processOperations(results, targetStore);
+  }
+
+  /** Returns true if the store id is equal to the local asset store's id */
+  private boolean isLocalAssetStoreId(String storeId) {
+    return getLocalAssetStore().getStoreType().equals(storeId);
+  }
+
+  /** Returns true if the store id is not equal to the local asset store's id */
+  private boolean isRemoteAssetStoreId(String storeId) {
+    return !isLocalAssetStoreId(storeId);
+  }
+
+  @Override public Opt<Asset> getAsset(Version version, final String mpId, final String mpeId) {
```

Comment:
This format is used in a few places in the AM already.  Personally I don't like it but was going for code that fits the style of the relevant codebase.  I'm going to fix it here regardless, since it helps with readability imo.